tripy/tripy/flat_ir/flat_ir.py

```diff
@@ -352,22 +352,11 @@ def register_tensor(self, tensor: "FlatIRTensor") -> "FlatIRTensor":
         return tensor
 
     def _get_constant_key(self, op):
-        from mlir_tensorrt.runtime._mlir_libs._api import MemRefValue
-        from tripy.utils.utils import list_to_tuple, volume
+        from mlir_tensorrt.runtime.api import MemRefValue
+        from tripy.utils.utils import list_to_tuple
 
         if isinstance(op.data, MemRefValue):
-            from tripy.backend.mlir.memref import tolist
-
-            VOLUME_THRESHOLD_FOR_MEMREF = 50
-            if volume(op.data.shape) < VOLUME_THRESHOLD_FOR_MEMREF:
-                l = tolist(op.data)
-            else:
-                l = [op.data.ptr]
-            data = list_to_tuple(l if isinstance(l, List) else [l])
-        elif isinstance(op.data, int) or isinstance(op.data, float) or isinstance(op.data, bool):
-            data = list_to_tuple(
-                op.data,
-            )
+            data = (op.data.ptr,)
```

Comment:
Won't this prevent us from deduplicating small constants?

Reply:
It will, but practically, if constant data is given by `MemrefValue`, it comes from users (such as weights & inputs). The small constants we want to deduplicate are generated internally where data is a list.

Reply:
Ah makes sense. Could you add a comment here to that effect?

Reply:
Added, thanks!